sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/numberFormatExpressions.scala

```diff
@@ -168,3 +168,142 @@ case class TryToNumber(left: Expression, right: Expression)
       newRight: Expression): TryToNumber =
     copy(left = newLeft, right = newRight)
 }
+
+/**
+ * A function that converts decimal values to strings, returning NULL if the decimal value fails to
+ * match the format string.
+ */
+@ExpressionDescription(
+  usage = """
+    _FUNC_(numberExpr, formatExpr) - Convert `numberExpr` to a string based on the `formatExpr`.
+      Throws an exception if the conversion fails. The format follows the same semantics as the
+      to_number function.
```

Comment:
shall we copy-paste the func doc from `to_number` and adjust it? I think we still need to explain some details, like `9` pads with spaces and `0` pads with `0`.

Reply:
Good idea, done. Making some edits helps the reader understand how each part of the result string is generated (as opposed to consumed).